src/msw/mediactrl_am.cpp

```diff
@@ -1590,11 +1600,10 @@ void wxAMMediaEvtHandler::OnActiveX(wxActiveXEvent& event)
             // size of the video (will error) - however on 4
             // it won't play on downloaded things until it is
             // completely downloaded so we use the lesser of two evils...
-            else if(event[0].GetInteger() == 3 &&
+            else if(event[0].GetInteger() >= 3 &&
```

Comment:
The comment above explicitly says that we don't want to do it on 4, so at the very least it needs to be changed/updated to avoid confusion. But it's not clear at all whether you decided it wasn't the lesser evil, finally, or if you think this problem doesn't exist any more or something else. Please explain it.

Reply:
The problem was that the size of the video could not be determined if the load event is sent on 3. This is solved by in essence delaying the sending of the load event. When the video finished loading, the AM software sends a stop event. The "workaround" takes advantage of this and sends the finish load event if it's been triggered by receiving event 3.
tl;dr: the FinishLoad has been broken down into a two part operation to avoid the problem with not being able to determine the video size.